src/State/helpers.js

```diff
@@ -25,3 +27,92 @@ export const transformRequests = (requests, role) => {
     }
   });
 };
+
+/* constructs a request config, then passing that to aws4 to sign
+ * host: the base url of the API
+ * method: the HTTP method of the request
+ * path: the specific path to be appended to the host (e.g. /users/requests)
+ * params: any query paramters the request needs. Pass in null if there is no query string parameters
+ * data: the body of the request. Pass in null if there is no body
+ */
+export const constructRequest = (host, method, path, params, data) => {
+  const {
```

Comment:
You can use `selectors` to get state from Selector folder. Is there any specific reason your using `getState`

Reply:
I have to access the the current state of the store outside of React components. calling `getState()` allows access to the state without creating a more complex redux middleware, or trying to use a saga selector.

Reply:
No not saga selectors. More of Redux selectors.  The reason I said that Cuz you have 2 to getState() in 2 different function. You can make it more modular. But this is not a big deal.

Reply:
i can probably pass in the access key, secret key, and session token into the signing method to reduce the number of calls to getState. I can see where it could run into errors. Good catch 👍

Reply:
👍